src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -132,4 +140,51 @@ void ColumnBetweenTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnBetweenTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter,
+                                                      const OrderByMode order_by_mode) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    using ColumnDataType = typename decltype(type)::type;
+
+    if constexpr (std::is_same_v<std::decay_t<decltype(typed_segment)>, ReferenceSegment>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
```

Comment:
@bakoe since this function is only called from _scan_non_reference_segment within this file (whose caller resolves the references already, see here and here) this here is fine to assume, I think...

Reply:
I see – can be considered resolved, to my mind. 🙂